frontend/app/src/components/table-filter/TableFilter.vue

```diff
@@ -279,12 +284,23 @@ function getDisplayValue(suggestion: Suggestion) {
 
 function getSuggestionText(suggestion: Suggestion) {
   const operator = suggestion.exclude ? '!=' : '=';
-  return `${suggestion.key}${operator}${getDisplayValue(suggestion)}`;
+  const startSelection = suggestion.key.length + operator.length;
+  const value = getDisplayValue(suggestion);
+  return {
+    text: `${suggestion.key}${operator}${value}`,
+    startSelection,
+    endSelection: startSelection + value.length,
+  };
 }
 
 function selectItem(suggestion: Suggestion) {
   nextTick(() => {
-    set(search, getSuggestionText(suggestion));
+    const suggestionText = getSuggestionText(suggestion);
+    set(search, suggestionText.text);
+    nextTick(() => {
+      get(input)?.$refs?.textInput?.focus();
+      get(input)?.$refs?.textInput?.setSelectionRange?.(suggestionText.startSelection, suggestionText.endSelection);
```

Comment:
This too feels like something we might want to expose as a public API if we are using it.

Reply:
Okay will it to my queue later